go/vt/wrangler/reparent.go

```diff
@@ -947,26 +1010,9 @@ func (wr *Wrangler) emergencyReparentShardLocked(ctx context.Context, ev *events
 		return fmt.Errorf("lost topology lock, aborting: %v", err)
 	}
 
-	// Verify masterElect is alive and has the most advanced position
-	masterElectStatus, ok := statusMap[masterElectTabletAliasStr]
-	if !ok {
-		return fmt.Errorf("couldn't get master elect %v replication position", topoproto.TabletAliasString(masterElectTabletAlias))
-	}
-	masterElectPos, err := mysql.DecodePosition(masterElectStatus.Position)
-	if err != nil {
-		return fmt.Errorf("cannot decode master elect position %v: %v", masterElectStatus.Position, err)
-	}
-	for alias, status := range statusMap {
-		if alias == masterElectTabletAliasStr {
-			continue
-		}
-		pos, err := mysql.DecodePosition(status.Position)
-		if err != nil {
-			return fmt.Errorf("cannot decode replica %v position %v: %v", alias, status.Position, err)
-		}
-		if !masterElectPos.AtLeast(pos) {
-			return fmt.Errorf("tablet %v is more advanced than master elect tablet %v: %v > %v", alias, masterElectTabletAliasStr, status.Position, masterElectStatus.Position)
-		}
+	// Bail out if the master tablet candidate is not the farthest ahead.
```

Comment:
I didn't realize that we ignore errors from `StopReplicationAndGetStatus()` above. Doesn't that mean ERS might cause you to lose transactions? I guess we need a force mode like that, but I would have expected an intermediate option as well that says, reparent only if it's safe.

@deepthi @sougou Did you know about this?

I'm also worried that it doesn't look like we have any step to wait for replicas to finish playing their relay logs. It seems like we just stop them and leave them stopped if our checks fail? If that's the case, then we might never meet the new criterion of ensuring the new master candidate has executed at least as far as anybody's relay log.